third_party/terraform/resources/resource_compute_network_peering.go.erb

```diff
@@ -64,18 +86,21 @@ func resourceComputeNetworkPeeringCreate(d *schema.ResourceData, meta interface{
 		return err
 	}
 
-	request := &compute.NetworksAddPeeringRequest{
-		Name:             d.Get("name").(string),
-		PeerNetwork:      d.Get("peer_network").(string),
-		AutoCreateRoutes: d.Get("auto_create_routes").(bool),
-	}
+	request := &computeBeta.NetworksAddPeeringRequest{}
+	<% unless version == 'ga' -%>
+		request.NetworkPeering = expandNetworkPeering(d)
+	<% else -%>
+	    request.Name 				=  d.Get("name").(string),
+ 	    request.PeerNetwork 		=  d.Get("peer_network").(string),
+	    request.AutoCreateRoutes    =  d.Get("auto_create_routes").(bool),
+	<% end -%>
 
-	addOp, err := config.clientCompute.Networks.AddPeering(networkFieldValue.Project, networkFieldValue.Name, request).Do()
+	addOp, err := config.clientComputeBeta.Networks.AddPeering(networkFieldValue.Project, networkFieldValue.Name, request).Do()
```

Comment:
Won't this cause that the beta endpoint to be called when using the GA provider?

Reply:
You're right but this happens throughout the GA provider - our provider distinction is more about features available than which API we hit.

Reply:
So we can leave it this way or should I add some more erb-conditionals? I recognized that too, but found similar behaviour in the whole provider.

Reply:
You can leave it.